src/utils/patterns.ts

```diff
@@ -210,24 +202,25 @@ const selectApplicationPineOptions = {
 			$select: 'slug',
 		},
 	},
-} satisfies PineOptions<Application>;
+} as const;
 
-type SelectApplicationResult = PineTypedResult<
-	Application,
-	typeof selectApplicationPineOptions
+type SelectApplicationResult = Pine.OptionsToResponse<
+	Application['Read'],
+	typeof selectApplicationPineOptions,
+	undefined
 >;
 
-export async function selectApplication(
-	filter?:
-		| PineFilter<Application>
-		| ((app: SelectApplicationResult) => boolean),
+export async function selectApplication<
+	T extends Pine.Filter<Application['Read']>,
+>(
+	filter?: T | ((app: SelectApplicationResult[number]) => boolean),
 	errorOnEmptySelection = false,
-) {
+): Promise<SelectApplicationResult[number]> {
 	const balena = getBalenaSdk();
 	let apps = (await balena.models.application.getAllDirectlyAccessible({
 		...selectApplicationPineOptions,
 		...(filter != null && typeof filter === 'object' && { $filter: filter }),
-	})) as SelectApplicationResult[];
+	})) satisfies SelectApplicationResult;
```

Comment:
As far as I tried it locally, this isn't necessary at all any more
```suggestion
	}));
```

Reply:
Yep! Thanks!